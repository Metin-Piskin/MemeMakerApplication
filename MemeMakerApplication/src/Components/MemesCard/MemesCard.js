import React from 'react';
import { Text, Image, StyleSheet, TouchableOpacity, Dimensions } from 'react-native';

const MemesCard = ({ veri, onPress }) => {
    return (
        veri ? (
            <TouchableOpacity
                style={styles.container}
                testID='memes-card-press'
                onPress={onPress}
            >
                <Image
                    source={{ uri: veri.url }}
                    testID='memes-card-image'
                    style={{
                        maxWidth: veri.width,
                        width: Dimensions.get('screen').width / 2.3,
                        maxHeight: veri.height,
                        height: Dimensions.get('screen').width / 2.3,
                        resizeMode: 'contain',
                        margin: 7,
                        //borderRadius: 20
                    }}
                />
            </TouchableOpacity>
        ) : (
            <Text>Veri Yok</Text>
        )
    )
}

export default MemesCard;

const styles = StyleSheet.create({
    container: {
        borderBottomWidth: 1,
        borderColor: '#fff',
        borderRadius: 10,
        //marginLeft: 2,
        marginTop: 7,
    },
})