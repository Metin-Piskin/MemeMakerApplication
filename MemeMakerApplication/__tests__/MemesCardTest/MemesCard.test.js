import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import MemesCard from '../../src/Components/MemesCard';

test('renders correctly', () => {
    const comp = render(<MemesCard />);
    expect(comp).toMatchSnapshot();
});

test('renders correctly', () => {
    const test = 'test'
    const comp = render(<MemesCard veri={test} />);
    expect(comp).toMatchSnapshot();
});

test('should render press correctly', () => {
    const test = 'test';
    const mockFunction = jest.fn();
    const comp = render(<MemesCard veri={test} onPress={mockFunction} />);

    const MemesCardPress = comp.getByTestId('memes-card-press');
    fireEvent(MemesCardPress, 'press');

    expect(mockFunction).toBeCalled();
})