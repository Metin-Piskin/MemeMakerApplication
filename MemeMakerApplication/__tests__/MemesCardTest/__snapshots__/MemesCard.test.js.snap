// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text>
  Veri Yok
</Text>
`;

exports[`renders correctly 2`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderBottomWidth": 1,
      "borderColor": "#fff",
      "borderRadius": 10,
      "marginTop": 7,
      "opacity": 1,
    }
  }
  testID="memes-card-press"
>
  <Image
    source={
      {
        "uri": undefined,
      }
    }
    style={
      {
        "height": 326.0869565217392,
        "margin": 7,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "resizeMode": "contain",
        "width": 326.0869565217392,
      }
    }
    testID="memes-card-image"
  />
</View>
`;
